File: packages/jsii-pacmak/test/expected.jsii-calc/java/src/main/java/software/amazon/jsii/tests/calculator/compliance/EnumDispenser.java
package software.amazon.jsii.tests.calculator.compliance;

/**
 * EXPERIMENTAL
 */
@javax.annotation.Generated(value = "jsii-pacmak")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.calculator.$Module.class, fqn = "jsii-calc.compliance.EnumDispenser")
public class EnumDispenser extends software.amazon.jsii.JsiiObject {

    protected EnumDispenser(final software.amazon.jsii.JsiiObjectRef objRef) {
        super(objRef);
    }

    protected EnumDispenser(final software.amazon.jsii.JsiiObject.InitializationMode initializationMode) {
        super(initializationMode);
    }

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public static @org.jetbrains.annotations.NotNull software.amazon.jsii.tests.calculator.compliance.AllTypesEnum randomIntegerLikeEnum() {
        return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.calculator.compliance.EnumDispenser.class, "randomIntegerLikeEnum", software.amazon.jsii.tests.calculator.compliance.AllTypesEnum.class);
    }

    /**
     * EXPERIMENTAL
     */
    @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
    public static @org.jetbrains.annotations.NotNull software.amazon.jsii.tests.calculator.compliance.StringEnum randomStringLikeEnum() {
        return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.calculator.compliance.EnumDispenser.class, "randomStringLikeEnum", software.amazon.jsii.tests.calculator.compliance.StringEnum.class);
    }
}

File: packages/jsii-pacmak/test/expected.jsii-calc/kotlin/src/main/kotlin/software/amazon/jsii/tests/kotlin/calculator/EnumDispenser.kt
package software.amazon.jsii.tests.kotlin.calculator

@javax.annotation.Generated("jsii-pacmak")
@software.amazon.jsii.Jsii(module = software.amazon.jsii.tests.kotlin.calculator.`$Module`::class, fqn = "jsii-calc.EnumDispenser")
@software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
open class EnumDispenser : software.amazon.jsii.JsiiObject {
    companion object {
        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun randomIntegerLikeEnum(): software.amazon.jsii.tests.kotlin.calculator.AllTypesEnum {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.EnumDispenser::class.java, "randomIntegerLikeEnum", software.amazon.jsii.tests.kotlin.calculator.AllTypesEnum::class.java) ?: error("Method 'randomIntegerLikeEnum()' returned null value")
        }

        @software.amazon.jsii.Stability(software.amazon.jsii.Stability.Level.Experimental)
        fun randomStringLikeEnum(): software.amazon.jsii.tests.kotlin.calculator.StringEnum {
            return software.amazon.jsii.JsiiObject.jsiiStaticCall(software.amazon.jsii.tests.kotlin.calculator.EnumDispenser::class.java, "randomStringLikeEnum", software.amazon.jsii.tests.kotlin.calculator.StringEnum::class.java) ?: error("Method 'randomStringLikeEnum()' returned null value")
        }
    }
    constructor(objRef: software.amazon.jsii.JsiiObjectRef) : super(objRef)
    constructor(mode: software.amazon.jsii.JsiiObject.InitializationMode) : super(mode)
}
